ledger/participant-integration-api/src/main/scala/platform/apiserver/RateLimitingInterceptor.scala

```diff
@@ -75,6 +83,34 @@ private[apiserver] final class RateLimitingInterceptor(metrics: Metrics, config:
     }
   }
 
+  private def memoryOverloaded(fullMethodName: String): Either[String, Unit] = {
+    tenuredMemoryPool.fold[Either[String, Unit]](Right(())) { p =>
+      if (p.isCollectionUsageThresholdExceeded) {
+        // Based on a combination of JvmMetricSet and MemoryUsageGaugeSet
+        val poolBeanMetricPrefix = s"jvm_memory_usage_pools_${p.getName}"
+        val rpcStatus =
+          s"""
+             | The ${p.getName} collection  has exceeded the maximum (${p.getCollectionUsageThreshold}).
+             | The rejected call was $fullMethodName.
+             | Jvm memory metrics are available at $poolBeanMetricPrefix
+          """.stripMargin
+        gc()
```

Comment:
Could you add here a comment explaining why it is safe to call gc (likely repeatedly) here?